### DISCUSSION TITLE
Future of Python 3.7 and Google Colab

### DISCUSSION THREAD

**Discussion by ilia3101** [colour-science/colour#1043] [Q&A]
### Question

Tried to install [`feature/cam16_hke_hellwig2022`](https://github.com/colour-science/colour/pull/1041) in Google Colab, and realised colour now requries Python 3.8, and Colab is stuck on 3.7. I tried changing the version requirement in the [`pyproject.toml`](https://github.com/colour-science/colour/blob/develop/pyproject.toml), and it installed, but submodules don't seem to load.

Are there any work arounds?

And is there any plan on supporting Google Colab while it's on 3.7?

**Comment by thomasmansencal** [comment]
Hey @ilia3101,

The Google Colab situation is rather bad and there is a thread here that you might want to weight in: https://github.com/googlecolab/colabtools/issues/2165

I really do hope that they will not can it...

That being said, you should be able to install Colour has follows:

```python
import sys

!pip install colour-science
!pip install --upgrade numpy

!pip uninstall -y colour-science
!if [ -d "colour" ]; then rm -rf colour; fi
!git clone https://github.com/colour-science/colour && cd colour && git fetch && git checkout develop && cd ..;
sys.path.append('colour')

!python -c "import imageio;imageio.plugins.freeimage.download()"

sys.path.append('colour')
```

Here is a notebook running the latest: https://colab.research.google.com/drive/1OerRYxnKOYGhiZEZda1QS93JWBfN_WI0?usp=sharing

**Reply by ilia3101** [reply]
Thanks for linking the notebook!
